### PR TITLE
[HttpClient] Make HttplugClient implement PSR-17 factories instead of Httplug's

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

Nobody uses `php-http/message-factory` anyway:
https://packagist.org/packages/php-http/message-factory-implementation/dependents

### CODE REVIEW

File: UPGRADE-6.2.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ HttpFoundation
  * Deprecate `Request::getContentType()`, use `Request::getContentTypeFormat()` instead
  * Deprecate calling `JsonResponse::setCallback()`, `Response::setExpires/setLastModified/setEtag()`, `MockArraySessionStorage/NativeSessionStorage::setMetadataBag()`, `NativeSessionStorage::setSaveHandler()`   without arguments
 
+HttpClient
+----------
+
+ * Deprecate implementing `RequestFactory`, `StreamFactory` and `UriFactory` on `HttplugClient`
```
Comment (stof) [inline UPGRADE-6.2.md]: I would use the FQCN or make it clear that those are HTTPlug interfaces

Comment (nicolas-grekas) [reply UPGRADE-6.2.md]: updated, thanks for the review

File: src/Symfony/Component/HttpClient/HttplugClient.php [hunk 2]
```diff
@@ -57,7 +57,7 @@
  *
  * @author Nicolas Grekas <p@tchwork.com>
  */
-final class HttplugClient implements HttplugInterface, HttpAsyncClient, RequestFactory, StreamFactory, UriFactory, ResetInterface
+final class HttplugClient implements HttplugInterface, HttpAsyncClient, RequestFactoryInterface, StreamFactoryInterface, UriFactoryInterface, RequestFactory, StreamFactory, UriFactory, ResetInterface
```
Comment (stof) [inline src/Symfony/Component/HttpClient/HttplugClient.php:64]: shouldn't we add a check for `RequestFactoryInterface` before the class definition, as done for RequestFactory ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/HttplugClient.php:64]: good ideas, updated

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by MGatner** [comment]
It looks like this added `symfony/deprecation-contracts` to Composer's `provide` key instead of `require`, which is causing the package not to be installed in some cases because Composer thinks this dependency takes care of it 😢

**Comment by xabbuh** [comment]
FTR, the previous comment was addressed in #47873.

**Pull request merged**